sql/catalyst/src/main/scala/org/apache/spark/sql/AnalysisException.scala

```diff
@@ -43,6 +43,11 @@ class AnalysisException protected[sql] (
   }
 
   override def getMessage: String = {
+    val planAnnotation = plan.map(p => s";\n$p").getOrElse("")
```

Comment:
When hitting an `AnalysisException`, we called `getMessage` in `SQLQueryTestSuite.scala`. That means, we also output the logical plan in the `.sql.out` when hitting an `AnalysisException`. The logical plan contains expression IDs. Thus, when we doing the text compare between the query results with the contents in `.sql.out`, it might fail due to the mismatch of expression IDs. Thus, I added an extra function here. Maybe I need to add a comment to say `this is for testing only`.